test/e2e/buildfarm_test.go

```diff
@@ -0,0 +1,219 @@
+
+			// create empty farm
+			cmd = []string{"buildfarm", "create", "farm3"}
+			session = podmanTest.Podman(cmd)
+			session.WaitWithDefaultTimeout()
+			Expect(session).Should(Exit(0))
+			Expect(session.Out.Contents()).Should(ContainSubstring("Farm \"farm3\" created"))
+
+			cfg, err := config.ReadCustomConfig()
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(cfg.Engine.DefaultFarm).Should(Equal("farm1"))
+			Expect(cfg.Engine.Farms).Should(HaveKeyWithValue("farm1", []string{"QA", "QB"}))
+			Expect(cfg.Engine.Farms).Should(HaveKeyWithValue("farm2", []string{"QA"}))
+			Expect(cfg.Engine.Farms).Should(HaveKeyWithValue("farm3", []string{}))
+
+			// create existing farm should exit with error
+			cmd = []string{"buildfarm", "create", "farm3"}
+			session = podmanTest.Podman(cmd)
+			session.WaitWithDefaultTimeout()
+			Expect(session).Should(Not(Exit(0)))
+		})
+
+		It("update existing farms", func() {
+			// create farm with multiple system connections
+			cmd := []string{"buildfarm", "create", "farm1", "QA", "QB"}
+			session := podmanTest.Podman(cmd)
+			session.WaitWithDefaultTimeout()
+			Expect(session).Should(Exit(0))
+			Expect(session.Out.Contents()).Should(ContainSubstring("Farm \"farm1\" created"))
+
+			// create farm with only one system connection
+			cmd = []string{"buildfarm", "create", "farm2", "QA"}
+			session = podmanTest.Podman(cmd)
+			session.WaitWithDefaultTimeout()
+			Expect(session).Should(Exit(0))
+			Expect(session.Out.Contents()).Should(ContainSubstring("Farm \"farm2\" created"))
+
+			// create empty farm
+			cmd = []string{"buildfarm", "create", "farm3"}
+			session = podmanTest.Podman(cmd)
+			session.WaitWithDefaultTimeout()
+			Expect(session).Should(Exit(0))
+			Expect(session.Out.Contents()).Should(ContainSubstring("Farm \"farm3\" created"))
+
+			cfg, err := config.ReadCustomConfig()
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(cfg.Engine.DefaultFarm).Should(Equal("farm1"))
+			Expect(cfg.Engine.Farms).Should(HaveKeyWithValue("farm1", []string{"QA", "QB"}))
+			Expect(cfg.Engine.Farms).Should(HaveKeyWithValue("farm2", []string{"QA"}))
+			Expect(cfg.Engine.Farms).Should(HaveKeyWithValue("farm3", []string{}))
+
+			// update farm1 to remove the QA connection from it
+			cmd = []string{"buildfarm", "update", "--remove", "QA", "farm1"}
+			session = podmanTest.Podman(cmd)
+			session.WaitWithDefaultTimeout()
+			Expect(session).Should(Exit(0))
+			Expect(session.Out.Contents()).Should(ContainSubstring("Farm \"farm1\" updated"))
+
+			// update farm3 to add QA and QB connections to it
+			cmd = []string{"buildfarm", "update", "--add", "QA,QB", "farm3"}
+			session = podmanTest.Podman(cmd)
+			session.WaitWithDefaultTimeout()
+			Expect(session).Should(Exit(0))
+			Expect(session.Out.Contents()).Should(ContainSubstring("Farm \"farm3\" updated"))
+
+			// update farm2 to be the default farm
+			cmd = []string{"buildfarm", "update", "--default", "farm2"}
```

Comment:
What if `farm1` is already default and I do a `buildfarm update --default=false farm1`?

Reply:
Good question - if that is the case we set DefaultFarm to "" so that none of the existing farms are set as the default. 
Added a test case for this.